lib/rosette/tms/smartling-tms/translation_memory_downloader.rb

```diff
@@ -0,0 +1,80 @@
+# encoding: UTF-8
+
+require 'thread'
+require 'concurrent'
+require 'smartling/uri'
+require 'restclient'
+
+module Rosette
+  module Tms
+    module SmartlingTms
+      class TranslationMemoryDownloader
+
+        TMX_API_PATH = 'translations/download'
+        DEFAULT_THREAD_POOL_SIZE = 10
+
+        attr_reader :configurator
+
+        def initialize(configurator)
+          @configurator = configurator
+        end
+
+        def build
+          if configurator.thread_pool_size > 0
+            build_asynchronously
+          else
+            build_synchronously
+          end
+        end
+
+        private
+
+        def build_synchronously
+          repo_config.locales.each_with_object({}) do |locale, ret|
+            ret[locale.code] = download(locale)
+          end
+        end
+
+        def build_asynchronously
+          pool = Concurrent::FixedThreadPool.new(configurator.thread_pool_size)
+          hash_mutex ||= Mutex.new
```

Comment:
is this an inherited `attr_accessor` from somewhere? why the memoization? is it just a concurrency thing?